operator/cilium_node.go

```diff
@@ -468,3 +467,82 @@ func updateCNP(ciliumClient v2.CiliumV2Interface, cnp *cilium_v2.CiliumNetworkPo
 		}
 	}
 }
+
+func RunCNPStatusNodesCleaner(ctx context.Context) {
+	go clearCNPStatusNodes(ctx, false)
+	go clearCNPStatusNodes(ctx, true)
+}
+
+func clearCNPStatusNodes(ctx context.Context, clusterwide bool) {
+	body, err := json.Marshal([]k8s.JSONPatch{
+		{
+			OP:   "remove",
+			Path: "/status/nodes",
+		},
+	})
+	if err != nil {
+		log.WithError(err).Debug("Unable to json marshal")
+		return
+	}
+
+	continueID := ""
+	for {
+		if clusterwide {
+			ccnpList, err := ciliumK8sClient.CiliumV2().CiliumNetworkPolicies(core_v1.NamespaceAll).List(
+				ctx,
+				meta_v1.ListOptions{
+					Limit:    10,
+					Continue: continueID,
+				})
+			if err != nil {
+				log.WithError(err).Debug("Unable to list CCNPs")
+				break
+			}
+			continueID = ccnpList.Continue
+
+			for _, cnp := range ccnpList.Items {
+				_, err = ciliumK8sClient.CiliumV2().CiliumClusterwideNetworkPolicies().Patch(ctx,
```

Comment:
It is not possible to check with the JSONPatch like you have previously mentioned but we should only perform an API call to kube-apiserver if we know that `len(cnp.status.nodes) != 0`. Doing this for all CNPs can put pressure on the api-server unnecessarily.